BackendIntegrationTests/BackendIntegrationTests.swift

```diff
@@ -215,23 +215,62 @@ class BackendIntegrationSK1Tests: XCTestCase {
 
         expect(customerInfo.entitlements.all.count) == 1
         let entitlements = self.purchasesDelegate.customerInfo?.entitlements
-        expect(entitlements?["premium"]?.isActive) == true
+        expect(entitlements?[Self.entitlementIdentifier]?.isActive) == true
 
         let anonUserID = Purchases.shared.appUserID
         let identifiedUserID = "\(#function)_\(anonUserID)_".replacingOccurrences(of: "RCAnonymous", with: "")
 
         let (identifiedCustomerInfo, created) = try await Purchases.shared.logIn(identifiedUserID)
         expect(created) == true
-        expect(identifiedCustomerInfo.entitlements["premium"]?.isActive) == true
+        expect(identifiedCustomerInfo.entitlements[Self.entitlementIdentifier]?.isActive) == true
 
         eligibility = await Purchases.shared.checkTrialOrIntroDiscountEligibility([productID])
         expect(eligibility[productID]?.status) == .ineligible
     }
 
+    func testUserHasNoEligibleOffersByDefault() async throws {
+        let (_, created) = try await Purchases.shared.logIn(UUID().uuidString)
+        expect(created) == true
+
+        let offerings = try await Purchases.shared.offerings()
+        let product = try XCTUnwrap(offerings.current?.monthly?.storeProduct)
+
+        expect(product.discounts).to(haveCount(1))
+        expect(product.discounts.first?.offerIdentifier) == "com.revenuecat.monthly_4.99.1_free_week"
+
+        let offers = await product.getEligiblePromotionalOffers()
+        expect(offers).to(beEmpty())
+    }
+
+    func testPurchaseWithDiscount() async throws {
```

Comment:
should this now be called `testPurcahseWithPromotionalOffer`?

Reply:
Yup fixed.